src/js/core/player.js

```diff
@@ -792,17 +792,77 @@ function name(): {country: string, firstName: string, lastName: string} {
+        const mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        const middleNameRow = playerNames.first[country].find(row => row[1] >= mnRand);
+        if (middleNameRow === undefined) {
+            throw new Error(`Undefined middleNameRow (mnRand=${mnRand})`);
+        }
+        let middleName = middleNameRow[0];
+
+        // abbreviation rules:
+        // 25%% will initialize their middle name
+        //   e.g. {Xavier Y. Zed}
+        // 25%% will use it fully but as a 'nickname'
+        //   e.g. {Xavier 'Yanick' Zed}
+        const abbrev = random.uniform(0, 1);
+        if (abbrev < 0.25 || country === "USA") {
+            // initialize it
+            middleName = `${middleName.charAt(0)}.`;
+        } else if (abbrev < 0.5) {
+            // use it fully
+            middleName = `'${middleName}'`;
+        }
+        firstName = `${firstName} ${middleName}`;
     }
-    const firstName = firstNameRow[0];
 
     // Last name
     const lnRand = random.uniform(0, playerNames.last[country][playerNames.last[country].length - 1][1]);
     const lastNameRow = playerNames.last[country].find(row => row[1] >= lnRand);
     if (lastNameRow === undefined) {
-        throw new Error(`Undefined lastNameRow (lnRand=${lnRand}`);
+        throw new Error(`Undefined lastNameRow (lnRand=${lnRand})`);
+    }
+    let lastName = lastNameRow[0];
+
+    // patronym names are not based on actually finding Dad
+    // we go up to XXX (30th) to outlast the popes,
+    // who currently set the record with John XXIII (23rd)
+    // however, we only do it in English-speaking countries
+    // we use a base rate of 2% of all players
+    const patBase = 0.02;
+    const patronymRand = random.uniform(0, 1);
+    const patNames = ["Jr.", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X", "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII", "XIX", "XX", "XXI", "XXII", "XXIII", "XXIV", "XXV", "XXVI", "XXVII", "XXVIII", "XXIX", "XXX"];
+    const patPercents = [patBase];
+    let sonContinue = 0.1; // 10% will name their kid III
+    while (patPercents.length < patNames.length) {
+        // III rate increases but tops out at 60%
+        if (patPercents.length <= 6) {
+            sonContinue = patPercents.length / 10.0;
+        }
+        patPercents.push(sonContinue * Math.min(...patPercents));
+    }
+    const english = ["USA", "Canada", "Australia", "New Zealand", "England", "Ireland", "Jamaica", "Phillippines"];
+    if (english.includes(country)) {
+        // we start at a base rate of 2% of all players
+        // then drill until we find out which patronym
+        if (patronymRand < patBase) {
+            const patNameRow = patPercents.findIndex(row => row <= patronymRand) - 1;
+            if (patNameRow < 0) {
+                // patronymRand less than 1.89e-10
+                // patronym would be higher than 30th
+                lastName = `${lastName} the Unlucky`;
```

Comment:
Other options I mulled were "the Chosen One", "the Second Coming of His Airness", "the Prince Who Was Promised", "the Boy Who Lived", etc. Unlucky is a placeholder til we find a suitable funny joke because I'm not interested in building a roman numerals enumerator to test the player into the 1e-20 range, you know?

If I got to choose, I'd probably vote for `{firstName lastName}, the Boy Who Lived` or `{firstName lastName}, His Royal Airness` with no preference between the two. Hell, write them both in and flip a coin because that conditional branch is going to be so rare, what's it matter to do one more math.random call?